.github/workflows/protocol_compatibility.yaml

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2020-2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Backwards compatibility check
+on:
+  pull_request:
+    paths:
+      - "src/controller/data_model/controller-clusters.matter"
+
+jobs:
+  check_clusters_matter:
+    name: Check controller-clusters.matter backwards compatibility
+    runs-on: ubuntu-latest
+    if: github.event.pull_request && !(contains(github.event.pull_request.labels.*.name, 'skip-protocol-compatibility'))
+
+    steps:
+       - name: Checkout
+         uses: actions/checkout@v3
+       - name: Setup python
+         uses: actions/setup-python@v4
+         with:
+           python-version: 3.11
+       - name: Install dependencies
+         run: |
+           python -m pip install --upgrade pip
+           pip install click coloredlogs lark
+       - name: Create old/new copies
+         run: |
+           mkdir -p out
+           cp src/controller/data_model/controller-clusters.matter out/new_version.matter
+           wget '${{github.event.pull_request.diff_url}}' --output-document out/patch.diff
+           patch -p1 --reverse <out/patch.diff
+           cp src/controller/data_model/controller-clusters.matter out/old_version.matter
+           patch -p1 <out/patch.diff
```

Comment:
I wonder whether `git checkout master -- src/controller/data_model/controller-clusters.matter` might work better for getting the old version?

Reply:
This may have changed more files in the patch, so probably `git checkout -f .` would be correct ... but at the same time just making it symmetric does not seem too bad. The reason for the revert is in case we change scripts as well.

Overall I believe for now this works so I am weary of changing things. Will happily checkmark followups on this though if it turns out it is better in some way.

Reply:
Ah, I see about master (I was thinking about the revert at first) I was unsure if our checkout `--depth=1` actually pulls in any history at all. The patch seemed to be guaranteed and fast enough.

Reply:
For now I would say this works on branches too, so it may be good. I think events objects may give me more data as well (I believe I get the ref source) I am not sure how much of a hassle getting that working is, so going by "I know this works today". We can update to something cleaner (the only update one file and not patch the world is for sure nicer) as a followup.